aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -258,19 +272,46 @@ Tensor _bincount_cuda_template(
     AT_ERROR("input and weights should have the same length");
   }
 
-  auto nbins = self.max().item<int64_t>() + 1L;
-  nbins = std::max(nbins, minlength);
+  const int64_t nbins = std::max(*self.max().cpu().data<input_t>() + 1L, minlength);
+  const input_t minvalue = 0;
+  const input_t maxvalue = nbins;
   // alloc output counter on GPU
   Tensor output;
   if (has_weights) {
     output = native::zeros({nbins}, weights.options());
     auto ret = cuda::CUDA_tensor_histogram<weights_t, input_t, true>(
-        output, self, weights, nbins, 1);
+        output, self, weights, nbins, minvalue, maxvalue);
   } else {
     output = native::zeros({nbins}, device(DeviceType::CUDA).dtype(kLong));
     auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, false>(
-        output, self, weights, nbins, 1);
+        output, self, weights, nbins, minvalue, maxvalue);
+  }
+  return output;
+}
+
+///////////////// histc /////////////////
+template <typename input_t>
+Tensor _histc_cuda_template(
+    const Tensor& self,
+    int64_t nbins,
+    input_t min,
+    input_t max) {
+  if (nbins <= 0) {
+    AT_ERROR("bins must be > 0");
   }
+  Tensor output = native::zeros({nbins}, device(DeviceType::CUDA).dtype(kLong));
+  input_t minvalue = min;
+  input_t maxvalue = max;
+  if (min == max) {
+    minvalue = *self.min().cpu().data<input_t>();
+    maxvalue = *self.max().cpu().data<input_t>();
+  }
+  if (minvalue == maxvalue) {
+    minvalue = minvalue - 1;
+    maxvalue = maxvalue + 1;
+  }
+  auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, false>(
+    output, self, at::empty({0}, self.options()), nbins, minvalue, maxvalue);
```

Comment:
`at::empty({0}, self.options())` actually creates a tensor which is a some overhead that we can avoid by passing in an undefined tensor. You can create an undefined tensor by not initializing a tensor, as so:
```
Tensor undefined;
auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, false>(
     output, self, undefined, nbins, minvalue, maxvalue);
```